text/0000-refactor-sbom.md

```diff
@@ -0,0 +1,258 @@
+
+The `signer` will accept configuration that will allow it to attach attestations to the application image. Perhaps:
+
+```toml
+[image]
+reference = "<image digest>"
+
+[[attestations]]
+type = "<media type>"
+uri = "<path to file with contents>"
+value = "<some arbitrary value, to be used instead of uri>"
+
+    [[attestations.annotations]]
+    name = "<name>"
+    value = "<value>"
+```
+
+For each "launch" SBOM file output by buildpacks, the `exporter` will create attestation configuration, e.g.:
+
+`<layers>/sbom/launch/<buildpack-id>/<layer-id>/bom.<ext>` (where `<ext>` is one of: `cdx.json`, `spdx.json`, or `syft.json` as currently spec'd)
+
+or more concretely:
+
+### Example 1 - layer-scoped SBOM file
+
+The `exporter` will turn `/layers/sbom/launch/my-buildpack/launch-dep/sbom.syft.json` into configuration:
+
+```toml
+[[attestations]]
+type = "application/vnd.cyclonedx+json"
+uri = "/layers/sbom/launch/my-buildpack/launch-dep/bom.syft.json"
+
+    [[attestations.annotations]]
+    name = "io.buildpacks.buildpack.id"
+    value = "my-buildpack"
+    
+    [[attestations.annotations]]
+    name = "io.buildpacks.sbom.scope"
+    value = "layer=sha256:$DIGEST" # or maybe layer name? somewhere?
+```
+
+And the `signer` will turn this into an in-toto statement with `predicateType` `https://cyclonedx.org/bom`:
+
+TODO
+
+It's worth noting, many buildpacks-produced SBOM files will be "layer-scoped" meaning that they don't describe the whole image, just a layer within it.
+`cosign` has a [deprecated](https://github.com/sigstore/cosign/blob/main/specs/SBOM_SPEC.md#scopes) method of denoting the scope of an SBOM that is mirrored here.
+However there are probably newer recommended ways of doing this, perhaps within the predicate - this requires further research (TODO).
+
+### Example 2 - image-scoped SBOM file
+
+The `exporter` will turn `/layers/sbom/launch/my-buildpack/launch.sbom.cdx.json` into configuration:
+
+```toml
+[[attestations]]
+type = "application/vnd.cyclonedx+json"
+uri = "/layers/sbom/launch/my-buildpack/launch.sbom.cdx.json"
+
+    [[attestations.annotations]]
+    name = "io.buildpacks.buildpack.id"
+    value = "my-buildpack"
+    
+    [[attestations.annotations]]
+    name = "io.buildpacks.sbom.scope"
+    value = "all"
+```
+
+And the `signer` will turn this into an in-toto statement with `predicateType` `https://cyclonedx.org/bom`:
+
+TODO
+
+### Example 3 - platform-provided SBOM file
+
+Platforms could inject TOML to describe the run image, e.g.:
+
+```toml
+[[attestations]]
+type = "application/vnd.cyclonedx+json"
+uri = "/platform/some-file.cdx.json"
+    
+    [[attestations.annotations]]
+    name = "io.buildpacks.sbom.scope"
+    value = "run" # maybe?
+```
+
+Potentially, the lifecycle could translate the special value `"run"` into a list of layer digests such as: `layer=sha256:$DIGEST-1,layer=sha256:$DIGEST-2`
+to save work for the platform to produce a more meaningful scope.
```

Comment:
Or, the lifecycle could read attestations for the run image directly. Or, perhaps this should also be out of scope.